### ISSUE TITLE
Bug: Create&update node in transaction bug

### ISSUE THREAD

**Issue by acquamarin** [kuzudb/kuzu#3904]
### Kùzu version

master

### What operating system are you using?

MAC OS

### What happened?
Dataset:
```
kuzu> begin transaction;
-
(0 tuples)
(0 columns)
Time: 0.01ms (compiling), 0.68ms (executing)
kuzu> create node table person (id int64, primary key(id));
----------------------------------
| result                         |
----------------------------------
| Table person has been created. |
----------------------------------
(1 tuple)
(1 column)
Time: 0.20ms (compiling), 2.93ms (executing)
kuzu> create (p:person {id: 14});
-
(0 tuples)
(0 columns)
Time: 0.05ms (compiling), 0.11ms (executing)
kuzu> create (p:person {id: 24});
-
(0 tuples)
(0 columns)
Time: 0.11ms (compiling), 0.22ms (executing)
kuzu> match (p:person {id:24}) set p.id=44;
-
(0 tuples)
(0 columns)
Time: 0.29ms (compiling), 0.60ms (executing)
```
Two issues:
Issue1:
```
kuzu> match (p:person {id:14}) set p.id=44;
[1]    46515 segmentation fault  ./kuzu dsadadddasd
```
Issue2:
```
kuzu> match (p:person {id:14}) set p.id=24;
Error: Runtime exception: Found duplicated primary key value 24, which violates the uniqueness constraint of the primary key column.
kuzu> match (p:person) return p.*;
--------
| p.id |
--------
| 14   |
--------
| 44   |
--------
(2 tuples)
(1 column)
Time: 0.15ms (compiling), 0.17ms (executing)
```

**Comment by ray6080** [comment]
This should be solved by #3899.

**Comment by andyfengHKU** [comment]
This should be fixed by #3924
